### PR TITLE
Adding FastText Text Classification

### PR DESCRIPTION
**Pull request by sravanbabuiitm**
## Description ##
Adding FastText classification to gluonnlp

TODO :
1. Ran only on ag_news and sogou_news, need to run and verify numbers on all datasets.
2. clean up the code and change print stmts to logging 
3. remove dependency on sklearn and change everything to mxnet.

### Essentials ###
- [N ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: scripts/classification/FastTextGluon.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    print("Network initialized")
+
+    softmax_cross_entropy = gluon.loss.SoftmaxCrossEntropyLoss(sparse_label=False)    
+
+    num_epochs = args.epochs
+    batch_size = args.batch_size
+    print("Starting Training!")
+    learning_rate = args.lr
+    trainer1 = gluon.Trainer(net.embedding.collect_params(), 'adam', {'learning_rate': learning_rate})
+    trainer2 = gluon.Trainer(net.dense.collect_params(), 'adam', {'learning_rate': learning_rate})
```
Comment (leezu) [inline scripts/classification/FastTextGluon.py]: No need to use 2 trainers when using the same hyperparameters. You can just do:
`trainer = gluon.Trainer(net.collect_params(), 'adam', {'learning_rate': learning_rate})`

File: scripts/classification/FastTextGluon.py [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+from mxnet.contrib import text
+from mxnet import nd, autograd, gluon
+from mxnet.gluon import nn, rnn, Block, HybridBlock
+import re
+from sklearn.preprocessing import LabelEncoder
+from sklearn.preprocessing import LabelBinarizer
+import gluonnlp.data.batchify as btf
+import argparse
+import evaluation
+import logging
```
Comment (astonzhang) [inline scripts/classification/FastTextGluon.py]: Thanks for contributing!
There are several redundant or unused imports, such as autograd. Could you resolve it?

**Comment by mli** [comment]
Job PR-258/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-258/3/index.html

**Comment by szha** [comment]
@sravanbabuiitm thanks for the contribution. I made some minor edits to the description to utilize the PR task list feature on github. Once you're ready, you can go ahead and check off the items by clicking them.

**Comment by szha** [comment]
Next steps once you have verified all the results:
- [ ] add a readme in https://raw.githubusercontent.com/dmlc/gluon-nlp/master/scripts/index.rst.
- [ ] add commands to reproduce the experiments.
- [ ] add experiment log to https://github.com/dmlc/web-data, and link to the log in the script's readme (see nmt or language model scripts for example http://gluon-nlp.mxnet.io/scripts/index.html)

**Comment by mli** [comment]
Job PR-258/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-258/5/index.html

**Comment by mli** [comment]
Job PR-258/9 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-258/9/index.html

**Comment by mli** [comment]
Job PR-258/10 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-258/10/index.html

**Comment by sravanbabuiitm** [comment]
Thank you Sheng :) 
Thanks for all the help and suggestions all along!
